fs2/tests/test_chunking.py

```diff
@@ -0,0 +1,147 @@
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from unittest import TestCase
+
+import torch
+from everyvoice.config.shared_types import ContactInformation
+from everyvoice.tests.stubs import silence_c_stderr
+from pydub import AudioSegment
+
+from ..config import FastSpeech2Config, FastSpeech2TrainingConfig
+from ..prediction_writing_callback import get_synthesis_output_callbacks
+from ..type_definitions import SynthesizeOutputFormats
+from .dummy_vocoder import get_dummy_vocoder
+
+
+class ChunkingTestBase(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.contact = ContactInformation(
+            contact_name="Test Runner", contact_email="info@everyvoice.ca"
+        )
+        cls.output_key = "output"
+        cls.outputs = {
+            cls.output_key: torch.ones([2, 500, 80], device="cpu"),
+            "duration_prediction": torch.ones([2, 7], device="cpu"),
+            "tgt_lens": [490, 490],
+        }
+        cls.batch1 = {
+            "basename": ["one", "two"],
+            "raw_text": ["This is chunk one.", "This is chunk two."],
+            "text": [
+                torch.IntTensor([2, 3, 4, 5, 6, 7, 8], device="cpu"),
+                torch.IntTensor([2, 3, 4, 5, 6, 7, 8], device="cpu"),
+            ],
+            "speaker": [
+                "S1",
+                "S2",
+            ],
+            "language": [
+                "L1",
+                "L2",
+            ],
+            "last_input_chunk": [1, 0],
+        }
+        cls.batch2 = {
+            "basename": ["three", "four"],
+            "raw_text": ["This is chunk three.", "This is chunk four."],
+            "text": [
+                torch.IntTensor([2, 3, 4, 5, 6, 7, 8], device="cpu"),
+                torch.IntTensor([2, 3, 4, 5, 6, 7, 8], device="cpu"),
+            ],
+            "speaker": [
+                "S1",
+                "S2",
+            ],
+            "language": [
+                "L1",
+                "L2",
+            ],
+            "last_input_chunk": [0, 1],
+        }
+
+
+class TestWritingWav(ChunkingTestBase):
+    """
+    Testing chunking when writing wav files.
+    """
+
+    def test_wav_chunks(self):
+        """
+        Tests that:
+        1. The outputted file is named after the first chunk
+        2. last_file_written contains that filename
+        3. The concatenated outputs are the correct lengths
+        """
```

Comment:
these are good comments, but unfortunately, unittest only shows the first line in the logs.
This looks like when you run it:
```
test_wav_chunks (model.feature_prediction.FastSpeech2_lightning.fs2.tests.test_chunking.TestWritingWav.test_wav_chunks)
Tests that: ... ok
```

So in the `""" """` docstring comments, you should write a one-line summary of this test function, but then you can put the more detailed comments as regular comments instead of docstring comments.